MRN-Code/nodeapi
# Coinstac consortia ctd

# Asana

https://app.asana.com/0/45796090191289/48021793625791
# Discussion

I recommend reviewing this PR on by looking at the diff's for each commit. The commits are clearly labeled, and accomplish the following:
1. Add a `PUT consortia` endpoint
2. Modify the `consortia` object to have the property `dbUrl`, which is the URL that the database for that consortia can be found at.
3. Modify houchdb, our docs, and our config to work with Cloudant remote CouchDB services
4. Modify docs to explain how to use a regular CouchDB server (e.g. a locally hosted one)
5. Set up watches on consortium databases, and re-calculate a simple average whenever data changes.

The last item is quite crude, and is just a proof of concept at this point.
# Testing

There are now three possible COUCH database configurations for running tests:
1. Use PouchDB's Leveldown adapter to create a Pouch Client that writes to the HD.
2. Install CouchDB on your server
3. Create a Cloudant account

If you wish, you can test all three, but it is probably not necessary.

Note also, when using Cloudant, tests will sometimes fail, due to server-startup timeouts... Just re-run the tests, and all should be well.
